### PR TITLE
New request

### PR DESCRIPTION
**Pull request by Seminire**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -29,6 +32,8 @@ app.use((req, res, next) => {
 app.use("/api/users", usersRoutes.router);
 app.use("/api/admin", adminRoutes.router);
 app.use("/api/discussion", discussionRoutes.router);
+app.use("/api/events-admin", eventsRoutes.router);
+app.use("/api/events-user", eventsURoutes.router);
```
Comment (Sirlahm1) [inline app.js:36]: You dont need this route.. all your user related event logic should be on the user route that is already existing

File: routes/events-user.js
Comment (Sirlahm1) [inline routes/events-user.js:1]: all the logic in this file should be in user route

File: routes/events-routes.js
Comment (Sirlahm1) [inline routes/events-routes.js:1]: all the logic in this file should be in admin route

File: controller/events-controller.js
Comment (Sirlahm1) [inline controller/events-controller.js:1]: there should be a seperation of concern, all logic related to user should be put in user controller whereas the admin also

File: controller/events-controller.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+const getAllEvents = async (req, res) =>{
+        //console.log(events);
+        const events = await Event.find({});
+        res.status(200).json(events);
+}
+
+const getEventById = async (req, res) => {
+    let event;
+    try {
+      const FoundEvents = await Event.findById((event) => event.id ===id)
```
Comment (Sirlahm1) [inline controller/events-controller.js:31]: i think this is wrong.. did you test the code?

File: controller/events-controller.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    } catch (err) {
+      console.log(err);
+      return res
+        .status(500)
+        .json({ message: "Events Deleted." });
+    }
+  };
+
+
+const addEvent =  async (req,res) => {
```
Comment (Sirlahm1) [inline controller/events-controller.js:42]: This logic here is wrong

**Review by Sirlahm1 — COMMENTED** [review]
learn more about mongodb .. you are not getting the mongodb logic

File: app.js (hunk 1 above)
Comment (Seminire) [reply app.js:36]: okay noted

File: controller/events-controller.js (hunk 2 above)
Comment (Seminire) [reply controller/events-controller.js:31]: yes i did. The code is running

File: controller/admin-controller.js [hunk 4]
```diff
@@ -313,6 +315,68 @@ const downloadPaymentReport = async (req, res, next) => {
   }
 };
 
+//For the admin to be able to add events for the user
+const getAllEvents = async (req, res) =>{
+  const events = await Event.find({event_id: req.event.id});
```
Comment (Sirlahm1) [inline controller/admin-controller.js:320]: if you are getting all event .. you are not suppose to pass anything inside of find method

Comment (Sirlahm1) [inline controller/admin-controller.js:320]: if you are getting all event .. you are not suppose to pass anything inside of find method

File: models/events.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
+const mongoose = require("mongoose");
+const Schema = mongoose.Schema;
+
+const EventsSchema = new Schema({
+    event_id:{
```
Comment (Sirlahm1) [inline models/events.js:5]: you dont need this .. mongodb will automatically give an id

**Pull request closed without merging**